dashboard/Makefile

```diff
@@ -19,18 +19,33 @@ DETACHED_MODE := $(if $(DETACHED),-d,)
 NODE_VERSION :=`cat ../.node-version`
 PWD := $(shell pwd)
 
+TEMPLATE_LOCATION := ./application/lib/ood_core/batch_connect/templates
+TURBOVNC_TEMPLATE := turbovnc.rb
+KVM_TEMPLATE      := kvm.rb
+
 remote-dev clean-remote-dev:     LOGIN_HOST   := login.rc.fas.harvard.edu
 remote-dev clean-remote-dev:     APP_FOLDER   := ./fasrc/dev/dashboard
+remote-dev:                      TEMPLATE_URL := https://vdi.rc.fas.harvard.edu/pun/sys/dashboard/files/fs/opt/ood/ondemand/root/usr/share/gems/2.7/ondemand/2.0.12/gems/ood_core-0.17.1/lib/ood_core/batch_connect/templates
+
 remote-fasse clean-remote-fasse: LOGIN_HOST   := fasselogin.rc.fas.harvard.edu
 remote-fasse clean-remote-fasse: APP_FOLDER   := ./.fasseood/dev/dashboard
+remote-fasse:                    TEMPLATE_URL := https://fasseood.rc.fas.harvard.edu/pun/sys/dashboard/files/fs/opt/ood/ondemand/root/usr/share/gems/2.7/ondemand/2.0.16/gems/ood_core-0.17.6/lib/ood_core/batch_connect/templates
+remote-fasse:                    HTTP_PROXY   := http://rcproxy.rc.fas.harvard.edu:3128
+remote-fasse:                    HTTPS_PROXY  := http://rcproxy.rc.fas.harvard.edu:3128
 
 dev: version down up
 
 remote-dev remote-fasse: version
-	$(ENV) docker-compose build
-	echo "For FASSE, you need to be connected to the VPN"
+	$(ENV) http_proxy=$(HTTP_PROXY) https_proxy=$(HTTPS_PROXY) docker-compose build
+	@echo "You need to be connected to the VPN"
 	ssh $(REMOTE_USERNAME)@$(LOGIN_HOST) mkdir -p $(APP_FOLDER)
 	$(ENV) docker-compose run --no-deps --entrypoint="" ood su - ood bash -c "export PATH=/home/ood/bin:/home/ood/ondemand/dev/dashboard/node_modules/.bin:$$PATH; cd /home/ood/ondemand/dev/dashboard; scl enable rh-ruby27 rh-nodejs12 'gem install --user-install bundler -v 2.1.4 && bundle config --local path 'vendor/bundle' && bundle install && npm install yarn --save && bundle exec rake assets:precompile'"
+	@echo "You need to be connected to the VPN"
+	@read -p "OOD password to copy templates:" ood_password; \
+	echo "Downloading: $(TEMPLATE_URL)/$(TURBOVNC_TEMPLATE)"; \
+	curl -kL $(TEMPLATE_URL)/$(TURBOVNC_TEMPLATE) --user $(REMOTE_USERNAME):$$ood_password -o $(TEMPLATE_LOCATION)/$(TURBOVNC_TEMPLATE) ; \
+	echo "Downloading: $(TEMPLATE_URL)/$(KVM_TEMPLATE)"; \
+	curl -kL $(TEMPLATE_URL)/$(KVM_TEMPLATE) --user $(REMOTE_USERNAME):$$ood_password -o $(TEMPLATE_LOCATION)/$(KVM_TEMPLATE) ; \
```

Comment:
OOD uses basic auth, this requires an specific `Authorization` header.
There are no auth cookies being set in the response.

I tested the examples in stackoverflow, just in case, but they do not work in this case.

Reply:
Yes, you are right, we are not yet using OIDC in production. Even if there were session cookies, we would still need to send the `Authorization` header. I think for now you will just need to prompt for the password multiple times, since it is not secure to put in in plaintext into the process table, and it would be possibly even less secure to read it plaintext from a file.